lib/undercover/plugin.rb

```diff
@@ -17,32 +17,60 @@ module Danger
   class DangerUndercover < Plugin
     VALID_FILE_FORMAT = '.txt'
     DEFAULT_PATH = 'coverage/undercover.txt'
+    MAX_INLINE_COMMENTS = 30
+    MAX_CHARACTERS_PER_MESSAGE = 60_000
 
-    # Checks the file validity and warns if no file is found
-    # if a valid file is found then if there are no changes,
-    # shows the report as a message in Danger.
-    # If there are reports then it shows the report as a warning in danger.
+    # Checks the file validity and fails if no file is found.
+    # If a valid file is found, the content of it will be parsed according to options
+    # and reports will be sent to danger.
+    #
+    # Options
+    #   - *undercover_path* the path to the undercover.txt file.
+    #     Defaults to `coverage/undercover.txt`.
+    #   - *sticky* from (danger)[https://danger.systems/reference.html].
+    #     Defaults to `true`.
+    #   - *in_line* when `true` will report each line or block missing coverage as an in-line comment.
+    #     Defaults to `false`.
+    #   - *report_danger* when `true` will report a failure to danger when new code is at 0 test coverage.
+    #     Defaults to `false`, so it generates warnings instead.
+    #   - *max_inline_comments* determine the maximum number of reported messages.
```

Comment:
That sounds like a great potential contribution to `danger/danger`. I assume you include it here to save some time.

Reply:
yes, potentially :D